src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -46,6 +46,21 @@ let didPerformWorkStackCursor : StackCursor<boolean> = createCursor(false);
 // pushed the next context provider, and now need to merge their contexts.
 let previousContext : Object = emptyObject;
 
+if (__DEV__) {
+  var _isProcessingChildContext = false;
+  var onBeginProcessingChildContext = function() {
+    _isProcessingChildContext = true;
+  };
+  exports.onBeginProcessingChildContext = onBeginProcessingChildContext;
+  var onEndProcessingChildContext = function() {
+    _isProcessingChildContext = false;
+  };
+  exports.onEndProcessingChildContext = onEndProcessingChildContext;
+  exports.isProcessingChildContext = function() : boolean {
+    return _isProcessingChildContext;
+  };
+}
```

Comment:
onBeginProcessingChildContext/onEndProcessingChildContext is over/early abstracted. Just set the flag to true and false like we do with ReactCurrentOwner. Seems like you can unify this with the render warning flag though.